### PR TITLE
Handle detected cluster to integration id mapping via provisioner node only

### PR DESCRIPTION
**Pull request by r0h4n**
tendrl-bug-id: Tendrl/node-agent#728

### CODE REVIEW

File: tendrl/node_agent/node_sync/sds_detect.py [hunk 1]
```diff
@@ -37,49 +38,64 @@ def sync():
...
-                            NS.tendrl_context.cluster_name = sds_details.get(
-                                'detected_cluster_name')
-                            NS.tendrl_context.sds_name = sds_details.get(
-                                'pkg_name')
-                            NS.tendrl_context.sds_version = sds_details.get(
-                                'pkg_version')
-                            NS.tendrl_context.save()
+
+                                
+                        while True:
```
Comment (shtripat) [inline tendrl/node_agent/node_sync/sds_detect.py]: How do we make sure the above job is completed or failed? Should we wait indefinitely here?

Comment (r0h4n) [reply tendrl/node_agent/node_sync/sds_detect.py]: Yes, the result of this job is crucial for the cluster, so we should wait indefinitely

**Comment by shtripat** [comment]
Also while identifying a new peer in cluster and when we fire a import, we should raise an event I feel, that the detected node is auto imported.

**Comment by r0h4n** [comment]
@shtripat That event should be triggered at the end of import job for new nodes or during ExpandCluster flow

**Comment by shtripat** [comment]
I feel better we have at the end of expand cluster flow because we are sure, expand has successfully happened by that time.

File: tendrl/node_agent/node_sync/sds_detect.py [hunk 2]
```diff
@@ -42,39 +42,39 @@ def sync():
                         integration_index_key = \
                             "indexes/detected_cluster_id_to_integration_id/" \
                             "%s" % sds_details['detected_cluster_id']
-
-                        if "provisioner/%s" % NS.tendrl_context.integration_id \
-                            in NS.node_context.tags:
+                        _ptag = "provisioner/%s" % \
+                            NS.tendrl_context.integration_id
+                        if _ptag in NS.node_context.tags:
```
Comment (shtripat) [inline tendrl/node_agent/node_sync/sds_detect.py]: @GowthamShanmugam is the issue mentioned around this condition resolved? Do we need some fix here?

**Pull request merged**